libs/sdk-core/src/breez_services.rs

```diff
@@ -702,7 +702,17 @@ impl BreezServices {
 
         // update node state and channels state
         self.persister.set_node_state(&new_data.node_state)?;
+
+        let channels_before_update = self.persister.list_channels()?;
         self.persister.update_channels(&new_data.channels)?;
+        let channels_after_update = self.persister.list_channels()?;
+
+        // Fetch the static backup if needed and persist it
+        if channels_before_update.len() != channels_after_update.len() {
```

Comment:
If a channel is closed the static backup is not updated here, right?

Maybe comparing a hash or a HashSet of before/after channels list might work.

Reply:
That's right if a channel is closed the static backup is not fetched . This is the expected behavior as we should only fetch it when a new channel is opened, see docs here: https://docs.corelightning.org/docs/backup-and-recovery#static-channel-backup

Reply:
Ok, I thought the static backup only contains active channels